### ISSUE TITLE
Make filename retrieval function more generic

### ISSUE THREAD

**Issue by kripod** [TomerAberbach/postcss-fontpie#7]
In some occasions mentioned in #6, the `src` within `@font-face` declarations can’t be used for mapping to a filename.

I’d propose adding a more generic `resolveFilename` callback as a replacement for `srcUrlToFilename`, accepting parameters such as `src` and `fontFamily`:

```js
const postcssFontpie = require("postcss-fontpie");

module.exports = {
  plugins: [
    postcssFontpie({
      fontTypes: {
        "Atkinson Hyperlegible": "sans-serif",
        "JetBrains Mono": "mono",
      },
      resolveFilename: ({ src: _unused, fontFamily }) => {
        const fontFamilyNormalized = fontFamily.replace(" ", "-").toLowerCase();
        return require.resolve(
          `@fontsource/${fontFamilyNormalized}/files/${fontFamilyNormalized}-latin-400-normal.woff2`,
        );
      },
    }),
  ],
};

```

**Comment by kripod** [comment]
Please note that this would make implementing #6 unnecessary.

**Comment by TomerAberbach** [comment]
Thanks for the feedback! I think I like this option better than #6. I'll look into implementing something like this and will probably create a minor release with the `srcUrlToFilename` option deprecated

**Comment by TomerAberbach** [comment]
@kripod, mind taking a look at #8 and confirming it would solve your issue?

**Comment by kripod** [comment]
Yeah, that looks awesome, thank you!

I really appreciate your quick response and implementation 🚀

**Comment by TomerAberbach** [comment]
No problem! Just published in 1.1.0 😄
